extensions/guacamole-auth-sso/modules/guacamole-auth-sso-openid/src/main/java/org/apache/guacamole/auth/openid/token/TokenValidationService.java

```diff
@@ -201,4 +214,53 @@ public Set<String> processGroups(JwtClaims claims) throws GuacamoleException {
         // Could not retrieve groups from JWT
         return Collections.emptySet();
     }
+
+    /**
+     * Parses the given JwtClaims, returning the attributes contained
+     * therein, as defined by the attributes claim type given in
+     * guacamole.properties. If the attributes claim type is missing or
+     * is invalid, an empty set is returned.
+     *
+     * @param claims
+     *     A valid JwtClaims to extract groups from.
+     *
+     * @return
+     *     A Map of String,String representing the attributes and values
+     *     from the OpenID provider point of view, or an empty Map if
+     *     claim is not valid or the attributes claim type is missing.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties could not be parsed.
+     */
+    public Map<String, String> processAttributes(JwtClaims claims) throws GuacamoleException {
+        String attributesClaim = confService.getAttributesClaimType();
+
+        if (claims != null && !attributesClaim.isEmpty()) {
+            try {
+                String[] attrs = DELIMITER_PATTERN.split(attributesClaim);
+                logger.debug("Iterating over attributes claim list : {}", attrs.toString());
+                Map<String, String> tokens = new HashMap<String, String>(attrs.length);
+                for (String key: attrs) {
+                    String oidcAttr = claims.getStringClaimValue(key);
+                    if (oidcAttr != null && !oidcAttr.isEmpty()) {
+                            String tokenName = TokenName.canonicalize(key, OIDC_ATTRIBUTE_TOKEN_PREFIX);
+                            tokens.put(tokenName, oidcAttr.toString());
+                            logger.debug("Claim {} found and set to {} as {}", key, tokenName, oidcAttr.toString());
+                    } else {
```

Comment:
Please do not cuddle the `else`. See established code and: https://guacamole.apache.org/guac-style/#braces